astroquery/ipac/irsa/most/core.py

```diff
@@ -0,0 +1,1017 @@
+        """
+        Parses the response when output type is set to `Regular` or `Full`.
+
+        Parameters
+        ----------
+        response : `requests.models.Response`
+            Query response.
+        withTarballs : `bool`, optional
+            Parse the links to FITS and region tarballs from the response. By
+            default, set to False.
+
+        Returns
+        -------
+        retdict : `dict`
+            Dictionary containing the keys `results`, `metadata` and `region`.
+            Optionally can contain keys `fits_tarball` and `region_tarball`.
+            The `results` and `metadata` are an `astropy.table.Table` object
+            containing the links to image and region files and minimum object
+            metadata, while `metadata` contains the image metadata and object
+            positions. The `region` key contains a link to the DS9 region file
+            representing the matched object trajectory and search boxes. When
+            existing, `fits_tarball` and `region_tarball` are links to the
+            tarball archives of the fits and region images.
+        """
+        retdict = {}
+        html = BeautifulSoup(response.content, "html5lib")
+        download_tags = html.find_all("a", string=re.compile(".*Download.*"))
+
+        # this is "Download Results Table (above)"
+        results_response = self._request("GET", download_tags[0]["href"])
+        retdict["results"] = Table.read(results_response.text, format="ipac")
+
+        # this is "Download Image Metadata with Matched Object position Table"
+        imgmet_response = self._request("GET", download_tags[1]["href"])
+        retdict["metadata"] = Table.read(imgmet_response.text, format="ipac")
+
+        # this is "Download DS9 Region File with the Orbital Path", it's a link
+        # to a DS9 region file
+        # regions_response = self._request("GET", download_tags[2]["href"])
+        retdict["region"] = download_tags[2]["href"]
+
+        if withTarballs:
+            retdict["fits_tarball"] = download_tags[-1]["href"]
+            retdict["region_tarball"] = download_tags[-2]["href"]
+
+        return retdict
+
+    @class_or_instance
+    def list_catalogs(self):
+        """Returns a list of queriable catalogs."""
+        response = self._request("GET", conf.interface_url, timeout=self.TIMEOUT)
+
+        html = BeautifulSoup(response.content, "html5lib")
+        catalog_dropdown_options = html.find("select").find_all("option")
+
+        catalogs = [tag.string for tag in catalog_dropdown_options]
+        # I think I saw somewhere that some password prompt will pop up for
+        # catalogs listed as '--- Internal use only:' but there are seemingly
+        # no limits to queries there.
+        if "--- Internal use only:" in catalogs:
+            catalogs.remove("--- Internal use only:")
+        return catalogs
+
+    def get_images(
+            self,
+            catalog="wise_merge",
```

Comment:
mandate all optional kwargs to be kwarg only, here and in all the other methods, too (at minimum in the public ones).

Reply:
The way this reads to me is as if you want me to make optional keyword arguments just keyword arguments - in which case every invocation of the method will require the user to provide them and that sounds wrong? Alternative understanding is to user `**kwargs` but that sounds wrong too? Could you clarify what you meant?